subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/ComponentSelectorSerializer.java

```diff
@@ -84,12 +84,13 @@ ImmutableVersionConstraint readVersionConstraint(Decoder decoder) throws IOExcep
         String requires = decoder.readString();
         String prefers = decoder.readString();
         String strictly = decoder.readString();
+        String branch = decoder.readNullableString();
```

Comment:
@gradle/dependency-management does this look correct/appropriate?

We made the other changes in this PR and started getting a failure: https://builds.gradle.org/viewQueued.html?itemId=16488839

But I'm not sure why.